### PR TITLE
Lambda expressions / it-academy / JC1-110-22 / Evgeniy Antonovich / Done tasks

### PR DESCRIPTION
**Pull request by aevgeni**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class MathStatistics {
+    public List<Integer> getRandomNumbers(int amount, int from, int to) {
+        return new Random().ints(amount, from, to)
+                .boxed().collect(Collectors.toList());
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java:11]:
```suggestion
        return new Random().ints(amount, from, to)
                    .boxed()
                    .collect(Collectors.toList());
```

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    public int countEvenNumbers(Collection<Integer> collection) {
+        return (int) collection.stream().filter(x -> x % 2 == 0).count();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java:15]:
```suggestion
        return collection.stream()
                .filter(x -> x % 2 == 0)
                .count();
```

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class MathStatistics {
+    public List<Integer> getRandomNumbers(int amount, int from, int to) {
+        return new Random().ints(amount, from, to)
+                .boxed().collect(Collectors.toList());
+    }
+
+    public int countEvenNumbers(Collection<Integer> collection) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java:14]:
```suggestion
    public long countEvenNumbers(Collection<Integer> collection) {
```

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelper.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.rakovets.course.java.core.practice.lambda_expressions;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CityHelper {
+    public List<String> getUniqueCities(Collection<String> collection) {
+        return collection.stream().distinct().collect(Collectors.toList());
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelper.java:9]:
```suggestion
        return collection.stream()
                .distinct()
                .collect(Collectors.toList());
```

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
+package com.rakovets.course.java.core.practice.lambda_expressions;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+class CityHelperTest {
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java:10]:
```suggestion
public class CityHelperTest {
```

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    private static List<String> cities = Arrays.asList
+            ("Minsk", "Minsk", "Grodno", "Grodno", "Gomel", "Vitebsk", "Mogilev", "Brest");
+    CityHelper cityHelper = new CityHelper();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java:13]:
```suggestion
    private CityHelper cityHelper;
    
    @BeforeEach
    voide initEach() {
         cityHelper = new CityHelper()
    }
```

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Test
+    void testGetUniqueCities() {
+        List<String> expected = List.of("Minsk", "Grodno", "Gomel", "Vitebsk", "Mogilev", "Brest");
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java:18]:
```suggestion
    List<String> cities = List.of("Minsk", "Minsk", "Grodno", "Grodno", "Gomel", "Vitebsk", "Mogilev", "Brest");

```

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/TvUtil.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
+package com.rakovets.course.java.core.practice.lambda_expressions.tv;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TvUtil {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/TvUtil.java:7]:
```suggestion
public class TvStore {
```

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/TvUtil.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    Collection<Tv> tvCollection;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/TvUtil.java:8]:
```suggestion
    private Collection<Tv> tvCollection;
```

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java (hunk 2 above)
Comment (aevgeni) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java]: fixed it

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java (hunk 3 above)
Comment (aevgeni) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatistics.java]: fixed it

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java (hunk 5 above)
Comment (aevgeni) [reply src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java]: fixed it

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java (hunk 6 above)
Comment (aevgeni) [reply src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java]: fixed it

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java (hunk 7 above)
Comment (aevgeni) [reply src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java:23]: fixed it

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/TvUtil.java (hunk 8 above)
Comment (aevgeni) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/TvUtil.java]: fixed it

**Pull request closed without merging**